index.js

```diff
@@ -0,0 +1,214 @@
+  const elem = e.target;
+  if (parent.classList.contains('buttonClick') || elem.classList.contains('buttonClick')) {
+    if (elem.classList.contains('CapsLock') || parent.classList.contains('CapsLock')) {
+      if (pressCaps) {
+        if (elem.tagName === 'DIV') elem.classList.remove('buttonClick');
+        else if (parent.tagName === 'DIV') parent.classList.remove('buttonClick');
+      }
+      pressCaps = (!pressCaps);
+    } else {
+      if (elem.tagName === 'DIV') elem.classList.remove('buttonClick');
+      else if (parent.tagName === 'DIV') parent.classList.remove('buttonClick');
+      if (elem.classList.contains('ShiftLeft') || parent.classList.contains('ShiftLeft') || elem.classList.contains('ShiftRight') || parent.classList.contains('ShiftRight')) pressShift = false;
+      if (elem.classList.contains('language') || parent.classList.contains('language')) {
+        document.querySelector('textarea').focus();
+        if (localStorage.languageOfKeyboard === 'ru') {
+          keyFill('eng');
+        } else {
+          keyFill('ru');
+        }
+      }
+    }
+  } else {
+    document.querySelectorAll('div').forEach((key) => key.classList.remove('buttonClick'));
+    document.querySelector('.CapsLock').classList.add('buttonClick');
+    document.querySelector('textarea').focus();
+  }
+});
+
+
+document.querySelector('body').addEventListener('keydown', (event) => {
+  cursor = document.querySelector('textarea').selectionStart;
+  objOfKeys.forEach((item) => {
+    if (item.code === event.code) {
+      document.querySelector(`.${item.code}`).classList.add('buttonClick');
+      if (item.code === 'ShiftLeft') {
+        changeLanguage[0] = true;
+      } else if (item.code === 'AltLeft') {
+        changeLanguage[1] = true;
+      } else {
+        changeLanguage[0] = false;
+        changeLanguage[1] = false;
+      }
+      if (item.code === 'ShiftLeft' || item.code === 'ShiftRight') pressShift = true;
+      enterSymbols(item);
+    }
+  });
+});
+
+
+document.querySelector('body').addEventListener('keyup', (event) => {
+  objOfKeys.forEach((item) => {
+    if (item.code === event.code) {
+      if (item.code === 'CapsLock') {
+        if (pressCaps) {
+          document.querySelector(`.${item.code}`).classList.remove('buttonClick');
+        }
+        pressCaps = !pressCaps;
+      } else {
+        document.querySelector(`.${item.code}`).classList.remove('buttonClick');
+        if (item.code === 'CapsLock') pressCaps = (!pressCaps);
+        if (item.code === 'ShiftLeft' || item.code === 'ShiftRight') pressShift = false;
+        if ((item.code === 'ShiftLeft' || item.code === 'AltLeft') && (changeLanguage[0] === true && changeLanguage[1] === true)) {
+          changeLanguage[0] = false;
+          changeLanguage[1] = false;
+          if (localStorage.languageOfKeyboard === 'ru') {
+            keyFill('eng');
+          } else {
+            keyFill('ru');
+          }
+        }
+      }
+    }
+  });
+});
```

Comment:
So, inside of the index.js you should store only main logic, only calls of the function or major checks